app/permissions.go

```diff
@@ -116,6 +142,25 @@ func (a *App) ImportPermissions(jsonl io.Reader) error {
 			return err
 		}
 
+		if schemeConveyor.Name == systemSchemeName {
+			for _, roleIn := range schemeConveyor.Roles {
+				dbRole, err := a.GetRoleByName(roleIn.Name)
+				if err != nil {
+					return errors.New(err.Message)
+				}
+				_, err = a.PatchRole(dbRole, &model.RolePatch{
+					Permissions: &roleIn.Permissions,
+				})
+				if err != nil {
+					for _, schemeID := range createdSchemeIDs {
```

Comment:
I think this clean up must be called for any return sentence returning an error, and some of the errors return don't clean up the created schemes.

Reply:
You're right it's safer.